From 7f4bde0a1c7889f088c704d441323fefd12acffc Mon Sep 17 00:00:00 2001
From: Robert Marshall <rmarshall@redhat.com>
Date: Mon, 27 Feb 2017 15:59:53 -0500
Subject: [PATCH] Initialize variable for ppc environment (#1285601)

Building on PPC does not default the value of an uninitialized integer
to zero which caused test failures.

Related: rhbz#1285601
---
 grubby.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/grubby.c b/grubby.c
index 9544212..7cd5c90 100644
--- a/grubby.c
+++ b/grubby.c
@@ -2259,6 +2259,8 @@ void setDefaultImage(struct grubConfig *config, int isAddingBootEntry,
 	/* check validity of existing default or first-entry-found
 	   selection */
 	if (defaultKernelPath) {
+                /* we must initialize this */
+                firstKernelEntryIndex = 0;
 		/* user requested first-entry-found */
 		if (!findEntryByPath(config, defaultKernelPath,
 				     prefix, &firstKernelEntryIndex)) {
-- 
2.9.3

